docs/get-started/quick_start.md

```diff
@@ -74,22 +87,36 @@ There is a utility function called `fhe.inputset(...)` for easily creating rando
 
 ## Compiling the function
 
-You can use the `compile` method of a `Compiler` class with an inputset to perform the compilation and get the resulting circuit back:
+You can use the `compile` method of the `Compiler` class with an inputset to perform the compilation and get the resulting circuit back:
 
 <!--pytest-codeblocks:skip-->
 ```python
+print(f"Compilation...")
 circuit = compiler.compile(inputset)
 ```
 
-## Performing homomorphic evaluation
+## Generating the keys
 
-You can use the `encrypt_run_decrypt` method of a `Circuit` class to perform homomorphic evaluation:
+You can use the `keygen` method of the `Circuit` class to generate the keys (public and private):
 
 <!--pytest-codeblocks:skip-->
 ```python
-homomorphic_evaluation = circuit.encrypt_run_decrypt(4, 4)
+print(f"Key generation...")
+circuit.keygen()
 ```
 
 {% hint style="info" %}
-`circuit.encrypt_run_decrypt(*args)` is just a convenient way to do everything at once. It is implemented as `circuit.decrypt(circuit.run(circuit.encrypt(*args)))`.
+If you don't call the key generation explicitly keys will be generated lazily when it needed.
 {% endhint %}
+
+## Performing homomorphic evaluation
+
+Now you can easily perform the homomorphic evaluation using the `encrypt`, `run` and `decrypt` methods of the `Circuit`:
+
+<!--pytest-codeblocks:skip-->
+```python
+print(f"Homomorphic evaluation...")
+encrypted_x, encrypted_y = circuit.encrypt(2, 6)
+encrypted_result = circuit.run(encrypted_x, encrypted_y)
+result = circuit.decrypt(encrypted_result)
+```
```

Comment:
@BourgerieQuentin  this section should be removed if you have already `encrypt`, `run` and `decrypt` in the example.